clinica/iotools/converters/aibl_to_bids/utils/clinical.py

```diff
@@ -168,263 +208,119 @@ def create_sessions_tsv_file(
-            fields_dataset.append(sessions_fields[i])
+    )[["BIDS CLINICA", f"{study} location", study]].dropna()
 
-    files_to_read: List[str] = []
-    sessions_fields_to_read: List[str] = []
-    for i in range(0, len(sessions_fields)):
-        # If the i-th field is available
-        if not pd.isnull(sessions_fields[i]):
-            # Load the file
-            file_to_read_path = clinical_data_dir / field_location[i]
-            files_to_read.append(glob.glob(str(file_to_read_path))[0])
-            sessions_fields_to_read.append(sessions_fields[i])
-
-    rids = pd.read_csv(files_to_read[0], dtype={"text": str}, low_memory=False).RID
-    unique_rids = list(set(rids))
-    for rid in unique_rids:
-        for file in files_to_read:
-            df = pd.read_csv(file, dtype={"text": str})
-            if len(df.columns) == 1:
-                df = pd.read_csv(file, sep=";", low_memory=False)
-
-            visit_code = df.loc[(df["RID"] == rid), "VISCODE"]
-
-            for field in sessions_fields_to_read:
-                if field in list(df.columns.values) and field == "MMSCORE":
-                    mm_score = df.loc[(df["RID"] == rid), field]
-                    mm_score[mm_score == -4] = "n/a"
-
-                elif field in list(df.columns.values) and field == "CDGLOBAL":
-                    cd_global = df.loc[(df["RID"] == rid), field]
-                    cd_global[
-                        cd_global == -4
-                    ] = "n/a"  # todo (LATER) : do that mapping later, same for other fields
-
-                elif field in list(df.columns.values) and field == "DXCURREN":
-                    dx_curren = df.loc[(df["RID"] == rid), field]
-                    dx_curren[dx_curren == -4] = "n/a"
-                    dx_curren[dx_curren == 1] = "CN"
-                    dx_curren[dx_curren == 2] = "MCI"
-                    dx_curren[dx_curren == 3] = "AD"
-
-                elif field in list(df.columns.values) and field == "EXAMDATE":
-                    exam_date = df.loc[(df["RID"] == rid), field]
-
-                elif field in list(df.columns.values) and field == "PTDOB":
-                    patient_date_of_birth = df.loc[(df["RID"] == rid), field]
-
-        exam_dates = _clean_exam_dates(
-            rid, exam_date.to_list(), visit_code.to_list(), clinical_data_dir
-        )
+    for bids_id in get_bids_subjs_list(bids_dir):
+        rid = int(bids_id_factory(study)(bids_id).to_original_study_id())
+        sessions = pd.DataFrame(
+            {"session_id": get_bids_sess_list(bids_dir / bids_id)}
+        ).set_index("session_id", drop=False)
 
-        if not patient_date_of_birth.empty:
-            age = _compute_ages_at_each_exam(
-                patient_date_of_birth.values[0], exam_dates
+        for _, row in specifications.iterrows():
+            df = pd.read_csv(
+                glob.glob(str(clinical_data_dir / row[f"{study} location"]))[0],
```

Comment:
Maybe something like this would be a bit more robust:

```python
try:
    df = pd.read_csv(next(clinical_data_dir.glob(row[f"{study} location"])), dtype=...)
except StopIteration:
    # do something if file doesn't exist...
```